cmd/admin_auth_ldap.go

```diff
@@ -24,182 +24,159 @@ type (
+			&cli.StringFlag{Name: "user-dn", Usage: "The user's DN.", Required: true},
 		},
-		Flags: ldapSimpleAuthCLIFlags,
 	}
+}
 
-	microcmdAuthUpdateLdapSimpleAuth = &cli.Command{
+func microcmdAuthUpdateLdapSimpleAuth() *cli.Command {
+	return &cli.Command{
 		Name:  "update-ldap-simple",
 		Usage: "Update existing LDAP (simple auth) authentication source",
-		Action: func(c *cli.Context) error {
-			return newAuthService().updateLdapSimpleAuth(c)
+		Action: func(ctx context.Context, cmd *cli.Command) error {
+			return newAuthService().updateLdapSimpleAuth(ctx, cmd)
+		},
+		Flags: []cli.Flag{
+			&cli.Int64Flag{Name: "id", Usage: "ID of authentication source", Required: true},
+			&cli.StringFlag{Name: "name", Usage: "Authentication name."},
+			&cli.BoolFlag{Name: "not-active", Usage: "Deactivate the authentication source."},
+			&cli.BoolFlag{Name: "active", Usage: "Activate the authentication source."},
+			&cli.StringFlag{Name: "security-protocol", Usage: "Security protocol name."},
+			&cli.BoolFlag{Name: "skip-tls-verify", Usage: "Disable TLS verification."},
+			&cli.StringFlag{Name: "host", Usage: "The address where the LDAP server can be reached."},
+			&cli.IntFlag{Name: "port", Usage: "The port to use when connecting to the LDAP server."},
+			&cli.StringFlag{Name: "user-search-base", Usage: "The LDAP base at which user accounts will be searched for."},
+			&cli.StringFlag{Name: "user-filter", Usage: "An LDAP filter declaring how to find the user record that is attempting to authenticate."},
+			&cli.StringFlag{Name: "admin-filter", Usage: "An LDAP filter specifying if a user should be given administrator privileges."},
+			&cli.StringFlag{Name: "restricted-filter", Usage: "An LDAP filter specifying if a user should be given restricted status."},
+			&cli.BoolFlag{Name: "allow-deactivate-all", Usage: "Allow empty search results to deactivate all users."},
+			&cli.StringFlag{Name: "username-attribute", Usage: "The attribute of the user’s LDAP record containing the user name."},
+			&cli.StringFlag{Name: "firstname-attribute", Usage: "The attribute of the user’s LDAP record containing the user’s first name."},
+			&cli.StringFlag{Name: "surname-attribute", Usage: "The attribute of the user’s LDAP record containing the user’s surname."},
+			&cli.StringFlag{Name: "email-attribute", Usage: "The attribute of the user’s LDAP record containing the user’s email address."},
+			&cli.StringFlag{Name: "public-ssh-key-attribute", Usage: "The attribute of the user’s LDAP record containing the user’s public ssh key."},
+			&cli.BoolFlag{Name: "skip-local-2fa", Usage: "Set to true to skip local 2fa for users authenticated by this source"},
+			&cli.StringFlag{Name: "avatar-attribute", Usage: "The attribute of the user’s LDAP record containing the user’s avatar."},
+			&cli.StringFlag{Name: "user-dn", Usage: "The user's DN."},
```

Comment:
These flags duplicate 3 times, IMO it's less maintainable than the old approach.